app/validators/date_validator.rb

```diff
@@ -1,12 +1,20 @@
 class DateValidator < ActiveModel::EachValidator
   def validate_each(record, attribute, value)
-    record.errors.add(attribute, error_message) unless parsable_date?(value)
+    if incorrect_format?(value)
```

Comment:
These should be single if conditions so that all validation errors are populated at once.